### DISCUSSION TITLE
Download progress and error messages

### DISCUSSION THREAD

**Discussion by lianyi1003** [Tyrrrz/YoutubeExplode#689] [General]
Hello, I am a novice in C#, would like to ask if it is possible to get the download progress? Also, the latest version throws a "system objectdisposedexception" error message.

**Comment by Sekoree** [comment]
Iirc the methods that handle stream download have an overload that accepts Progress<double> that reports back the progress while it's downloading.
You can pass Progress a method to call when the progress value changes in its constructor.

**Comment by lianyi1003** [comment]
Sorry, I still don't quite understand, could you please explain in more detail?

**Reply by Tyrrrz** [reply]
You can see how the progres reporting is implemented in the demo projects in this repo

**Comment by mohammadhoseinGhezelsofla** [comment]
Haven't answer for the main question, but I have same problem, I'm also getting object disposed exception in latest version, using net framework 4.6.1. I'm not sure about its my problem or not so can't open a new issue
